pyfolio/risk.py

```diff
@@ -0,0 +1,486 @@
+    gross_exposures = []
+    net_exposures = []
+
+    positions_wo_cash = positions.drop('cash', axis=1)
+    tot_gross_exposure = positions_wo_cash.abs().sum(axis=1)
+    tot_long_exposure = positions_wo_cash[positions_wo_cash > 0].sum(axis=1)
+    tot_short_exposure = positions_wo_cash[positions_wo_cash < 0] \
+        .abs().sum(axis=1)
+
+    for i in range(1, len(CAP_CUTOFFS)+1):
+        if i == len(CAP_CUTOFFS):
+            in_bucket = positions_wo_cash[caps >= CAP_CUTOFFS[-1]]
+        else:
+            in_bucket = positions_wo_cash[((caps <= CAP_CUTOFFS[i]) &
+                                           (caps >= CAP_CUTOFFS[i-1]))]
+
+        gross_bucket = in_bucket.abs().sum(axis=1).divide(tot_gross_exposure)
+        long_bucket = in_bucket[in_bucket > 0] \
+            .sum(axis=1).divide(tot_long_exposure)
+        short_bucket = in_bucket[in_bucket < 0] \
+            .sum(axis=1).divide(tot_short_exposure)
+        net_bucket = long_bucket.subtract(short_bucket)
+
+        gross_exposures.append(gross_bucket)
+        long_exposures.append(long_bucket)
+        short_exposures.append(short_bucket)
+        net_exposures.append(net_bucket)
+
+    return long_exposures, short_exposures, gross_exposures, net_exposures
+
+
+def plot_cap_exposures_longshort(long_exposures, short_exposures, ax=None):
+    '''
+    Plots outputs of compute_cap_exposures as area charts
+
+    Parameters
+    ----------
+    long_exposures, short_exposures : arrays
+        Arrays of long and short market cap exposures (output of
+        compute_cap_exposures).
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    colors = ['#FF9999', '#FFCC99', '#99FF99', '#99CCFF', '#CC99FF']
+
+    ax.stackplot(long_exposures[0].index, long_exposures,
+                 labels=CAP_NAMES, colors=colors, baseline='zero')
+    ax.stackplot(long_exposures[0].index, short_exposures,
+                 colors=colors, baseline='zero')
+    ax.axhline(0, color='k', linestyle='-')
+    ax.set_title('Long and Short Exposures to Market Caps')
+    ax.set_ylabel('Proportion of Long/Short Exposure in Market Cap Buckets')
+    ax.legend(loc='upper left', frameon=True, framealpha=0.5)
+
+    return ax
+
+
+def plot_cap_exposures_gross(gross_exposures, ax=None):
+    '''
+    Plots outputs of compute_cap_exposures as area charts
+
+    Parameters
+    ----------
+    gross_exposures : array
+        Arrays of gross market cap exposures (output of compute_cap_exposures).
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    colors = ['#FF9999', '#FFCC99', '#99FF99', '#99CCFF', '#CC99FF']
+
+    ax.stackplot(gross_exposures[0].index, gross_exposures,
+                 labels=CAP_NAMES, colors=colors, baseline='zero')
+    ax.axhline(0, color='k', linestyle='-')
+    ax.set_title('Gross Exposure to Market Caps')
+    ax.set_ylabel('Proportion of Gross Exposure \n in Market Cap Buckets')
+
+    return ax
+
+
+def plot_cap_exposures_net(net_exposures, ax=None):
```

Comment:
Why do we need separate plotting functions for all different types of exposures? Couldn't a single function just plot whatever is passed in?